### PR TITLE
fix: [PL-40631]: Fix shell script delegate startup failure

### PR DESCRIPTION
**Pull request by VikasMaddukuriHarness**
## Describe your changes
Current production delegate is not a bijou delegate. The part of code which is causing error in case of shell script delegate was introduced as part of bijou delegate. We are trying to create an apiClient using the default k8s config of the customer env. This will fail for shell script delegate if the default config is not valid.

Added a catch block to catch any exceptions while installing apiClientModule and just log the exception and proceed further.

Testing:
1. tested in local and changes looks fine. Even if we remove the install part all together, the delegate is working fine.
## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### CODE REVIEW

File: 260-delegate/src/main/java/io/harness/delegate/app/modules/platform/k8s/K8SRunnerModule.java [hunk 1]
```diff
@@ -24,7 +24,10 @@ protected void configure() {
      * We don't need to re-install this module in immutable delegate
      install(new DelegateDecryptionModule());
      */
-    install(new ApiClientModule());
+    final var isBijouDelegateEnabled = System.getenv().get("BIJOU_DELEGATE_ENABLED");
```
Comment (raghuAtWings) [inline 260-delegate/src/main/java/io/harness/delegate/app/modules/platform/k8s/K8SRunnerModule.java:27]: @xingchi-jin are we introducing another variable? can we not auto detect it?

Comment (xingchi-jin) [reply 260-delegate/src/main/java/io/harness/delegate/app/modules/platform/k8s/K8SRunnerModule.java:27]: @VikasMaddukuriHarness  have you tried try catch (Exception) here.

The outcome is upon exception, the k8s apiClient will not be initialized, which is the same outcome as your change made here (when the BIJOU_DELEGATE_ENABLED env is not set)

Comment (VikasMaddukuriHarness) [reply 260-delegate/src/main/java/io/harness/delegate/app/modules/platform/k8s/K8SRunnerModule.java]: yes, we can use try/catch as well. Modified the code. Please re-review. Thanks.

Comment (xingchi-jin) [reply 260-delegate/src/main/java/io/harness/delegate/app/modules/platform/k8s/K8SRunnerModule.java]: make sense.

**Review by xingchi-jin — APPROVED** [review]

**Comment by VikasMaddukuriHarness** [comment]
trigger uts

**Comment by VikasMaddukuriHarness** [comment]
trigger messagecheck

**Pull request merged**